test/unit/emailjs-imap-client-imap-test.js

```diff
@@ -38,16 +38,18 @@
             socketStub = sinon.createStubInstance(TCPSocket);
             sinon.stub(TCPSocket, 'open').withArgs(host, port).returns(socketStub);
 
-            client.connect(TCPSocket).then(() => {
+            var promise = client.connect(TCPSocket).then(() => {
                 expect(TCPSocket.open.callCount).to.equal(1);
 
                 expect(socketStub.onerror).to.exist;
                 expect(socketStub.onopen).to.exist;
                 expect(socketStub.onclose).to.exist;
                 expect(socketStub.ondata).to.exist;
-            }).then(done).catch(done);
+            });
 
             setTimeout(() => socketStub.onopen(), 0);
+
+            return promise;
```

Comment:
Yeah, I realized that this is extremely convenient when writing unit tests with promises. No more `done()` ;)